driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/gridfs/GridFSUploadPublisherImpl.java

```diff
@@ -227,6 +233,7 @@ private Mono<Long> createSaveChunksMono(final AtomicBoolean terminated) {
             AtomicLong lengthInBytes = new AtomicLong(0);
             AtomicInteger chunkIndex = new AtomicInteger(0);
             new ResizingByteBufferFlux(source, chunkSizeBytes)
+                    .contextWrite(sink.contextView())
                     .flatMap((Function<ByteBuffer, Publisher<InsertOneResult>>) byteBuffer -> {
                         if (terminated.get()) {
                             return Mono.empty();
```

Comment:
I don't know why the tracing in the repo example was happy here. I also don't fully understand how reactor propagates it 🤷 

So it appears to automatically be propogated within scope - but I have added your suggestion to make it explicit.

Reply:
The change seems to having been done only to the `Publisher`s created via `insertOne`, but "this `Mono`", i.e., the `Mono` on line 239 where I originally left the comment (that line is now 238) still does not have `contextWrite` called on it.

> it appears to automatically be propogated within scope

What do you mean by "scope" here?

Reply:
> a Context is tied to each Subscriber in a chain. It uses the Subscription propagation mechanism to make itself available to each operator, starting with the final subscribe and moving up the chain.

So that makes sense why our code broke that.